### PR TITLE
#Sn-32-Create Subscription

### PR DESCRIPTION
**Pull request by PivovarAndrew**

### CODE REVIEW

File: app/models/subscription.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#  index_subscriptions_on_subscriber_id                   (subscriber_id)
+#  index_subscriptions_on_subscriber_id_and_signatory_id  (subscriber_id,signatory_id) UNIQUE
+#
+# Foreign Keys
+#
+#  fk_rails_...  (signatory_id => users.id)
+#  fk_rails_...  (subscriber_id => users.id)
+#
+class Subscription < ApplicationRecord
+  belongs_to :subscriber, class_name: 'User'
```
Comment (mlshvch) [inline app/models/subscription.rb]: please, add `validate: true` both for subscriber and signatory to ensure they exist

**Review by mlshvch — CHANGES_REQUESTED** [review]

**Comment by ramantechart** [comment]
You need to rebase your branch from main

**Comment by ramantechart** [comment]
Please make sure to update our diagram

File: db/seeds.rb [hunk 2]
```diff
@@ -38,3 +39,8 @@
 end
 
 FactoryBot.create_list(:tag, GENERATE_TAGS_COUNT)
+FactoryBot.create_list(:subscription, GENERATE_SUBSCRIPTIONS_COUNT)
```
Comment (ramantechart) [inline db/seeds.rb:52]: please reuse users that were created on line 25

File: spec/models/post_tag_spec.rb [hunk 3]
```diff
@@ -59,8 +59,9 @@
     end
 
     it 'is not saved' do
-      expect { post_tags.save }.to
-      raise_error(ActiveRecord::RecordInvalid, "Validation failed: Profile can't be blank")
+      expect do
+        post_tags.save!
+      end.to.raise_error(ActiveRecord::RecordInvalid, 'Validation failed: Tag must exist')
```
Comment (ramantechart) [inline spec/models/post_tag_spec.rb]: `end.to raise_error`

File: spec/models/post_tag_spec.rb [hunk 4]
```diff
@@ -59,8 +59,9 @@
     end
 
     it 'is not saved' do
-      expect { post_tags.save }.to
-      raise_error(ActiveRecord::RecordInvalid, "Validation failed: Profile can't be blank")
+      expect do
+        post_tags.save!
```
Comment (ramantechart) [inline spec/models/post_tag_spec.rb]: this let is not defined

File: db/seeds.rb [hunk 5]
```diff
@@ -20,6 +20,8 @@
 
 GENERATE_TAGS_COUNT = 30
 GENERATE_POSTS_COUNT = 5
+GENERATE_SUBSCRIPTIONS_COUNT = 10
+GENERATE_PROFILE_COUNT = 6
 
 users = FactoryBot.create_list(:user, NUMBER_OF_USERS)
```
Comment (mlshvch) [inline db/seeds.rb:26]: My suggestion:
`users_for_subscription = users
user1 = users_for_subscription.sample
user2 = users_for_subscription.sample
while user2 == user1
  user2 = users_for_subscription.sample
end`

**Pull request merged**